From: Sjoerd Simons <sjoerd.simons@collabora.co.uk>
Date: Sun, 16 Feb 2020 22:02:35 +0100
Subject: um: vector: Avoid NULL ptr deference if transport is unset

When the transport option of a vec isn't set strncmp ends up being
called on a NULL pointer. Better not do that.

Signed-off-by: Sjoerd Simons <sjoerd.simons@collabora.co.uk>

Series-to: linux-um@lists.infradead.org
---
 arch/um/drivers/vector_kern.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/um/drivers/vector_kern.c b/arch/um/drivers/vector_kern.c
index 769ffbd..9d1ba26 100644
--- a/arch/um/drivers/vector_kern.c
+++ b/arch/um/drivers/vector_kern.c
@@ -177,6 +177,9 @@ static int get_transport_options(struct arglist *def)
 	int vec_tx = VECTOR_TX;
 	long parsed;
 
+	if (transport == NULL)
+		return -EINVAL;
+
 	if (vector != NULL) {
 		if (kstrtoul(vector, 10, &parsed) == 0) {
 			if (parsed == 0) {
